lib/terraform/group.rb

```diff
@@ -35,13 +36,13 @@ def hostnames
       data[:hostnames]
     end
 
-    def store_output_names(name:)
+    def store_output_name(name:)
       output_names.push name
     end
 
     def merge_attributes
```

Comment:
I think all of this complexity can be removed if `Verifier#resolve_attributes` just resolves the default attributes before the user attributes and `Group` allows the values to be overridden.